---
title: "Fusion Results"
author: "Sehrish Kanwal"
date: "01/10/2018"
output: html_document
---

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE)
```

### Required packages

```{r}
suppressMessages(library(kableExtra))
library(tidyverse)
```

### Pizzly filtering

```{r}
#read in the pizzly fusion calls
pizzly.fusions <- read.table(file = '~/Documents/UMCCR/data/fusions/pizzly-validation/MH17T001P013-oncofuse-test-flat-filtered.tsv', header = TRUE)

#read in the transcripts quantification file
quant <- read.table(file = '~/Documents/UMCCR/data/fusions/pizzly-validation/abundance.tsv', header = TRUE)

#sort and filter quantification file on tpm values. Currently filtering on quantiles. Selected 0.997 because that reduces the final fusion
#calls to the value we are interested in (~15)
quant.sorted.filtered <- filter(arrange(quant, desc(quant$tpm)), tpm >= (quantile(quant$tpm, 0.999)))

#initialize an empty dataframe
result <- data.frame()
result.list <- vector("list", length = nrow(pizzly.fusions))

#let's try using for loop for iterating over pizzly.fusions dataframe and get transcriptID and fusion gene pair information.
#can also filter quant.sorted.filtered$target_id to have only fusion gene target ids (that is two tracscripts instead of one-
#this will increase speed

for (row in 1:nrow(pizzly.fusions)){
  y <- strsplit(as.character(pizzly.fusions[row, 7]), "\\;")
  y <- unname(y)
  for (i in 1:length(y[[1]])){
    if (y[[1]][i] %in% quant.sorted.filtered$target_id){
      #val <- c(y[[1]][i], as.character(pizzly.fusions[row,1]), as.character(pizzly.fusions[row, 3]))
      #creating a new dataframe for the filtered pizzly results
      result <- rbind(result, data.frame(pizzly.fusions[row,]))
      #result.list[[row]] <- val
    }
  }
}


#remove duplicated values from result filtered using expression count(as multiple transcripts might support fusion between same gene)
deduped.result <- unique(result)

#Extract only those fusion genes that are in PMCC list
#read in the PMCC genes list
pmcc.genes <- read.table(file = '~/Documents/UMCCR/data/RNAseq-report/2018-02-16_PMCC_Panel_Genes.txt', header = FALSE)
result.pmcc <- data.frame()
for (row in 1:nrow(pizzly.fusions)){
  if(pizzly.fusions[row,1] %in% pmcc.genes$V1 | pizzly.fusions[row,3] %in% pmcc.genes$V1){
    #creating a new dataframe for extracting pizzly rows with pmcc gene hits
    result.pmcc <- rbind(result.pmcc, data.frame(pizzly.fusions[row,]))
  }
}

#sorting pizzly.fusions according to two new target dataframes identified i.e. 1- pizzly result filtering using expressions counts and 2- pmcc gene list

pizzly.fusions$order <- rownames(pizzly.fusions) %in% rownames(deduped.result) 
pizzly.fusions$order2 <- rownames(pizzly.fusions) %in% rownames(result.pmcc)

pizzly.fusions2 <- arrange(pizzly.fusions, -order, -order2) %>%
  select(-order,-order2)

kable(pizzly.fusions2) %>%
  kable_styling(font_size = 10) %>%
  #row_spec(c(as.numeric(rownames(deduped.result)),as.numeric(rownames(result.pmcc))), bold = T, color = "white", background = "#D7261E") %>%
  group_rows("Group 1", 1, nrow(deduped.result)) %>%
  group_rows("Group 2", (nrow(deduped.result)+1), (nrow(deduped.result) + nrow(result.pmcc))) %>%
  scroll_box(width = "100%", height = "200px")
```
